hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -384,7 +377,7 @@ protected static long averageBytesPerRecord(HoodieTimeline commitTimeline, Hoodi
               .fromBytes(commitTimeline.getInstantDetails(instant).get(), HoodieCommitMetadata.class);
           long totalBytesWritten = commitMetadata.fetchTotalBytesWritten();
           long totalRecordsWritten = commitMetadata.fetchTotalRecordsWritten();
-          if (totalBytesWritten > fileSizeThreshold && totalRecordsWritten > 0) {
+          if (totalBytesWritten > fileSizeThreshold && totalBytesWritten <= fileSizeUpper && totalRecordsWritten > 0) {
             avgSize = (long) Math.ceil((1.0 * totalBytesWritten) / totalRecordsWritten);
             break;
```

Comment:
What's the purpose of this limitation?

Reply:
to achieve a similar goal as https://github.com/apache/hudi/pull/6864.

this consolidated implementation avoids logic fragments and is also UT-friendly.

Reply:
But the `totalBytesWritten` is the size accumulated from all the written files, what the meaning of it to compare with a single file size threshold?

Reply:
you are correct and I misunderstood the `totalBytesWritten`.  

then there might be two options:
1. consider extra FilesInsert/FileUpdated 
2. filter out REPLACE commit within the `averageBytesPerRecord` function. 

which one do you prefer?

Reply:
I just updated the code to reflect option 2.